### PR TITLE
Add virtual method to extend the complex property serialization

### PR DESCRIPTION
**Pull request by xuzhg**
A lot of customers complain about the complex property serialization.
For example, they want to control the complex property serialization in some circumstances, for example, if its value is null, skip it.

This PR simply adds a virtual method that the customer can override to control the complex property serialization.

This is the same as the navigation property serialization.

### CODE REVIEW

File: src/Microsoft.AspNetCore.OData/Formatter/Serialization/ODataResourceSerializer.cs [hunk 1]
```diff
@@ -960,6 +958,59 @@ await serializer.WriteObjectInlineAsync(propertyValue, edmProperty.Type, writer,
...
 
+        /// <summary>
+        /// Creates the <see cref="ODataNestedResourceInfo"/> to be written while writing this dynamic complex property.
+        /// </summary>
+        /// <param name="propertyName">The dynamic property name.</param>
+        /// <param name="propertyValue">The dynamic property value.</param>
+        /// <param name="edmType">The edm type reference.</param>
+        /// <param name="resourceContext">The context for the complex instance being written.</param>
+        /// <returns>The nested resource info to be written. Returns 'null' will omit this serialization.</returns>
+        /// <remarks>It enables customer to get more controll by overriding this method. </remarks>
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Formatter/Serialization/ODataResourceSerializer.cs:969]:
```suggestion
        /// <remarks>It enables customer to get more control by overriding this method.</remarks>
```

File: src/Microsoft.AspNetCore.OData/Formatter/Serialization/ODataResourceSerializer.cs [hunk 2]
```diff
@@ -960,6 +958,59 @@ await serializer.WriteObjectInlineAsync(propertyValue, edmProperty.Type, writer,
...
+        }
+
+        /// <summary>
+        /// Creates the <see cref="ODataNestedResourceInfo"/> to be written while writing this complex property.
+        /// </summary>
+        /// <param name="complexProperty">The complex property for which the nested resource info is being created.</param>
+        /// <param name="pathSelectItem">The corresponding sub select item belongs to this complex property.</param>
+        /// <param name="resourceContext">The context for the complex instance being written.</param>
+        /// <returns>The nested resource info to be written. Returns 'null' will omit this complex serialization.</returns>
+        /// <remarks>It enables customer to get more controll by overriding this method. </remarks>
```
Comment (gathogojr) [inline src/Microsoft.AspNetCore.OData/Formatter/Serialization/ODataResourceSerializer.cs:992]:
```suggestion
        /// <remarks>It enables customer to get more control by overriding this method.</remarks>
```

**Review by gathogojr — APPROVED** [review]
:shipit:

**Comment by xuzhg** [comment]
@gathogojr Resolved the comments, please take a look again.

**Review by gathogojr — APPROVED** [review]

**Pull request merged**